angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
I found a new wrinkle. When the second type argument is `A extends T[] | null | undefined`, the callback parameter is typed as `{}`, irrespective of `--strictNullChecks`. The type inference use case, IMO the primary one, would be lost if this is merged as is.